app/uk/gov/hmrc/cbcr/services/AuditSubscriptionService.scala

```diff
@@ -47,19 +51,22 @@ class AuditSubscriptionService @Inject()(
         .toList
         .flatMap(CBCId.apply)
 
-    repo
-      .getSubscriptions("cbcId" -> Json.obj("$in" -> cbcIds))
-      .map(sd =>
-        sd.foreach(s =>
-          auditSubscriptionDetails(s).onComplete {
-            case Success(AuditResult.Success) =>
-              logger.info(s"Successfully audited SubscriptionDetails of CBCId ${s.cbcId.toString}")
-            case Success(AuditResult.Failure(msg, _)) =>
-              logger.warn(s"Unable to audit SubscriptionDetails of CBCId ${s.cbcId.toString} $msg")
-            case Success(AuditResult.Disabled) =>
-              logger.warn(s"Auditing is disabled for SubscriptionDetails of CBCId ${s.cbcId.toString}")
-            case Failure(e) => logger.warn(s"Audit failed to complete for CBCId ${s.cbcId.toString}, ${e.getMessage}")
-        }))
+    Await.result(
+      repo
+        .getSubscriptions(Filters.in("cbcId", cbcIds: _*))
+        .map(sd =>
+          sd.foreach(s =>
+            auditSubscriptionDetails(s).onComplete {
+              case Success(AuditResult.Success) =>
+                logger.info(s"Successfully audited SubscriptionDetails of CBCId ${s.cbcId.toString}")
+              case Success(AuditResult.Failure(msg, _)) =>
+                logger.warn(s"Unable to audit SubscriptionDetails of CBCId ${s.cbcId.toString} $msg")
+              case Success(AuditResult.Disabled) =>
+                logger.warn(s"Auditing is disabled for SubscriptionDetails of CBCId ${s.cbcId.toString}")
+              case Failure(e) => logger.warn(s"Audit failed to complete for CBCId ${s.cbcId.toString}, ${e.getMessage}")
+          })),
+      Duration(60, "second")
```

Comment:
Sorry I dont understand this `Duration(60, "second")`

Reply:
Await(future, Duration(60, "second")) will make the thread wait for the future to complete for up to 60 seconds or throw an exception and fail if it doesn't do that in time. This way we can make sure the checks run _after_ the future completes (rather than sometimes before and sometimes after depending on random factors).

Reply:
Yes I know that logic. my question is different.
Is there any document or confluence page to use only 60 seconds for this case?

Reply:
There's no guidance as to how long the timeout should be.

Reply:
ok, Im not 100% sure about that limit. Please have a word with Dave once.